core/primitives/src/child_trie.rs

```diff
@@ -183,15 +183,27 @@ impl ChildTrie {
 		&p[CHILD_STORAGE_KEY_PREFIX.len()..]
 	}
 	/// Constructor to use for building a new child trie.
-	/// This does not allow setting an existing `KeySpace`.
-	pub fn new(keyspace_builder: &mut impl KeySpaceGenerator, parent: &[u8]) -> Self {
-		let parent = Self::prefix_parent_key(parent);
-		ChildTrie {
-			keyspace: keyspace_builder.generate_keyspace(),
-			root: Default::default(),
-			parent,
-			extension: Default::default(),
-		}
+	///
+	/// By using a `KeySpaceGenerator` it does not allow setting an existing `KeySpace`.
+	/// If a new trie is created (see `is_new` method), the trie is not commited and
+	/// another call to this method will create a new trie.
+	/// 
+	/// This can be quite unsafe for user, so use with care (write new trie information
+	/// as soon as possible).
+	pub fn fetch_or_new(
+		keyspace_builder: &mut impl KeySpaceGenerator,
+		parent_fetcher: impl FnOnce(&[u8]) -> Option<Self>,
```

Comment:
this is for incoming multiple level trie ? because otherwise the only function to put here is runtime_io::child_trie no ?

Reply:
Not really, it is just to keep things generic (but yes it should be runtime_io::child_trie that is call for general srml module).

Reply:
Then maybe I would put a new method in externalities to have fetch_or_new without this argument ?
and maybe rename to something fetch_or_new_pending because it doesn't create a new child trie until you actually write on it.

Reply:
if empty child trie can't exist then we can't do better than that I think